source/contracts/reporting/Auction.sol

```diff
@@ -0,0 +1,239 @@
+
+    modifier onlyAuthorizedPriceFeeder {
+        require(authorizedPriceFeeders[msg.sender]);
+        _;
+    }
+
+    modifier initializeNewAuctionIfNeeded {
+        if (currentAuctionIndex != getAuctionIndexForCurrentTime()) {
+            initializeNewAuction();
+        }
+        _;
+    }
+
+    function initialize(IUniverse _universe) public beforeInitialized returns (bool) {
+        endInitialization();
+        require(_universe != address(0));
+        universe = _universe;
+        reputationToken = universe.getReputationToken();
+        initializationTime = controller.getTimestamp();
+        authorizedPriceFeeders[controller.owner()] = true;
+        manualRepPriceInAttoEth = Reporting.getAuctionInitialRepprice();
+        lastRepPrice = manualRepPriceInAttoEth;
+        repPrice = manualRepPriceInAttoEth;
+        currentRepPrice = manualRepPriceInAttoEth;
+        bootstrapMode = true;
+        return true;
+    }
+
+    function initializeNewAuction() public returns (bool) {
+        if (currentAuctionType == AuctionType.RECORDED) {
+            repPrice = currentRepPrice;
+        }
+        lastRepPrice = currentRepPrice;
+        currentAuctionType = getAuctionType();
+        uint256 _currentAuctionIndex = getAuctionIndexForCurrentTime();
+        require(currentAuctionType == AuctionType.UNRECORDED || currentAuctionType == AuctionType.RECORDED);
+        require(currentAuctionIndex != _currentAuctionIndex);
+        if (bootstrapped) {
+            bootstrapMode = false;
+        }
+        require(!bootstrapMode || currentAuctionType == AuctionType.UNRECORDED);
+        bootstrapped = true;
+
+        uint256 _auctionRepBalanceTarget = reputationToken.totalSupply() / Reporting.getAuctionTargetSupplyDivisor();
+        uint256 _repBalance = reputationToken.balanceOf(this);
+
+        if (_repBalance < _auctionRepBalanceTarget) {
+            reputationToken.mintForAuction(_auctionRepBalanceTarget.sub(_repBalance));
+        } else {
+            reputationToken.burnForAuction(_repBalance.sub(_auctionRepBalanceTarget));
+        }
+
+        initialAttoRepBalance = _auctionRepBalanceTarget;
+        initialAttoEthBalance = address(this).balance;
+
+        currentAttoRepBalance = initialAttoRepBalance;
+        currentAttoEthBalance = initialAttoEthBalance;
+
+        currentUpperBoundRepPriceInAttoEth = 0;
+        currentLowerBoundRepPriceInAttoEth = 0;
+
+        currentAuctionIndex = _currentAuctionIndex;
+
+        initialRepSalePrice = lastRepPrice.mul(Reporting.getAuctionInitialPriceMultiplier());
+        initialEthSalePrice = Reporting.getAuctionInitialPriceMultiplier().mul(10**36).div(lastRepPrice);
+        return true;
+    }
+
+    function tradeRepForEth(uint256 _attoEthAmount) public initializeNewAuctionIfNeeded returns (bool) {
```

Comment:
Style: I'm not a huge fan of using modifiers for this sort of thing.  I like to think of modifiers as preconditions, while this modifier is an action (potentially).

Reply:
I wanted to really call out where this was happening and how the new auction initialization works.

Reply:
Will just make it a function